### DISCUSSION TITLE
Unexpected behaviour on g.E("1", "2", "3")

### DISCUSSION THREAD

**Discussion by anheri** [Gremlinq/ExRam.Gremlinq#843] [General]
Hi, 
The g.E() operation has an overload accepting "params obj[] ids"
so you can do something like this: g.E("1", "2", "3", "4") and that is nice
and you can do this:
obj[] someEdgeIds = GetSomeEdgeIds();
g.E(someEdgeIds).Drop()    
so far, so good...

The problem arises when, someEdgeIds is empty...
which ends up being the same as: g.E().Drop()... which deletes all edges in the entire database....

Is this intentional?
To me at least it was not expected... and I end up destroying my database.

**Comment by danielcweber** [comment]
That is the Gremlin semantics of `V()` and `E()`.

**Reply by anheri** [reply]
yes... but that is not the semantics in the world of C#...
in a linq statement like: customers.Where(x => someCustomerIds.Contains(x)), if someCustomerIds is empty... the query would return 0 results, not all.

So I think maybe g.E(emptyArray), should throw an argument exception...

**Reply by danielcweber** [reply]
I agree that whenever an array is passed and the array is empty, it is probably never the intention of the caller to be treated like `V()`. It would be a breaking change however and will have to wait for the next major release.

**Reply by anheri** [reply]
ok, that sounds reasonable...

thanks for a great api :)
